tests/System.IO.Pipelines.Tests/WritableBufferWriterFacts.cs

```diff
@@ -59,6 +59,30 @@ public void SlicesSpanAndAdvancesAfterWrite()
         }
 
         [Fact]
+        public void ThrowsForInvalidParameters()
+        {
+            _buffer = _pipe.Writer.Alloc(1);
+            var initialLength = _buffer.Buffer.Length;
+
+            var writer = new WritableBufferWriter(_buffer);
+            var array = new byte[] { 1, 2, 3 };
+
+            writer.Write(array, 0, 0);
+            writer.Write(array, array.Length, 0);
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => writer.Write(array, -1, 0));
```

Comment:
Fixed it in here:
https://github.com/dotnet/corefxlab/pull/1401/commits/d52726144b5f94dfcba9e2302555a4b64806c00c